### PR TITLE
Fix timestamps on log files created before GPS lock

### PR DESCRIPTION
**Pull request by tridge**

### CODE REVIEW

**Comment by peterbarker** [comment]
```
../../libraries/AP_Logger/AP_Logger_File.cpp: In member function ‘void AP_Logger_File::_io_timer()’:
../../libraries/AP_Logger/AP_Logger_File.cpp:1070:32: error: aggregate ‘AP_Logger_File::_io_timer()::utimbuf t’ has incomplete type and cannot be defined
                 struct utimbuf t;
```

(windows)

File: libraries/AP_Logger/AP_Logger_File.cpp [hunk 1]
```diff
@@ -1058,6 +1063,18 @@ void AP_Logger_File::_io_timer(void)
         ::fsync(_write_fd);
         last_io_operation = "";
 #endif
+
+        if (_need_rtc_update) {
+            uint64_t utc_usec;
+            if (AP::rtc().get_utc_usec(utc_usec)) {
+                struct utimbuf t;
```
Comment (peterbarker) [inline libraries/AP_Logger/AP_Logger_File.cpp]: Suggest:

```
const struct utimbuf t{
    modtime: utc_usec / (1000UL * 1000UL),
    actime: t.modtime
};
```

As things currently stand it's not clear to the reader if this struct is fully initialised or contains garbage from the stack.

Comment (peterbarker) [reply libraries/AP_Logger/AP_Logger_File.cpp]: .... OK, suggestion is not quite right here...

Comment (tridge) [reply libraries/AP_Logger/AP_Logger_File.cpp]: fixed to be only ChibiOS

File: libraries/AP_Logger/AP_Logger_File.cpp [hunk 2]
```diff
@@ -1058,7 +1063,24 @@ void AP_Logger_File::_io_timer(void)
         ::fsync(_write_fd);
         last_io_operation = "";
 #endif
+
+#if CONFIG_HAL_BOARD == HAL_BOARD_CHIBIOS
+        // ChibiOS does not update mtime on writes, so if we opened
```
Comment (OXINARF) [inline libraries/AP_Logger/AP_Logger_File.cpp]: I'm surprised by this (it should do it on fsync, not writes, but anyway). Is that a ChibiOS issue or a FatFS problem? I tried searching for it but couldn't find anything.

**Comment by OXINARF** [comment]
@tridge By the way, why the new Logging label? We already have the Dataflash label (which we should probably rename).

File: libraries/AP_Logger/AP_Logger_File.cpp [hunk 3]
```diff
@@ -1058,7 +1063,24 @@ void AP_Logger_File::_io_timer(void)
...
+        // without knowing the time we should update it later
+        if (_need_rtc_update) {
+            uint64_t utc_usec;
+            if (AP::rtc().get_utc_usec(utc_usec)) {
+                struct utimbuf t {};
+                t.modtime = utc_usec / (1000UL * 1000UL);
+                t.actime = t.modtime;
+                // we ignore return on utime() as there is nothing useful we can do
+                (void)utime(_write_filename, &t);
```
[the remark this answers is not in the document]
Comment (tridge) [reply libraries/AP_Logger/AP_Logger_File.cpp]: the void was a hint that this was deliberate, not a warning fix

**Comment by tridge** [comment]
@OXINARF I didn't know we could rename labels :-)

**Pull request merged**